modules/storage/storage.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ExtraVolumePropagation defines a const that represent the propagation policy
+type ExtraVolumePropagation string
+// ExtraVolType represents a "label" that can be optionally added to the VolMounts
+// instance
+type ExtraVolType string
+// ServiceType identifies the Service, Group or instance (e.g. the backend) that
+// receives an Extra Volume that can potentially be mounted
+type ServiceType string
+
+const (
+	// PropagationEverywhere is used to define a propagation policy that allows
+	// to get the volumes mounted to all the OpenStack services
+	PropagationEverywhere ExtraVolumePropagation = "All"
+	// DBSync represents a common ServiceType defined by the OpenStack operators
+	// that keeps track of the DBSync pod
+	DBSync                ServiceType            = "DBSync"
+	// Compute represents a common ServiceType that can be translated into an
+	// external-data-plane related propagation policy
+	Compute               ServiceType            = "Compute"
+)
+
+// VolMounts is the data structure used to expose Volumes and Mounts that can
+// be added to a pod according to the defined Propagation policy
+type VolMounts struct {
+	// +kubebuilder:validation:Required
+	// Propagation defines which pod should mount the volume
+	Propagation  []ExtraVolumePropagation `json:"propagation,omitempty"`
```

Comment:
should not have omitempty if it is required

Reply:
It's "required" in the sense that from meta-operator  we need to propagate volumes to the right services (External propagation) and we might want to select, within the single operator, which pod can mount what (so we need this parameter in the CR definition), but given this structure can be used in different contexts (e.g. ansible-operator [1]) where
the assumption is to get all the defined extra volumes and mount them, we might want to avoid the parameter, hence I'm keeping it optional.

[1] https://github.com/openstack-k8s-operators/ansibleee-operator/pull/6/files#diff-b7931efcf77c2d3bcfe0479b278a92deaec40ae5a56ccd6067f52985c01d8ab7R23

Reply:
Yes, that works, I'll do in the next iteration.